### PR TITLE
[NFC] some minor code cleanup of getHostCPUName handling

### PR DESCRIPTION
**Pull request by vtjnash**
Try to minimize the places the code depends on this library function, since it is useful for generic code as long as LLVM is available, but we would also be happy with returning the text "native" here in most cases.

### CODE REVIEW

**Comment by gbaraldi** [comment]
I think llvm is always available in this sense, since we always link these libs in julia-internal, but LGTM

**Comment by vtjnash** [comment]
Only part of it is available (libSupport), and we want to try to ensure it could feasibly be replaced or disabled if someone wanted to (e.g. targeting WASM, embedded, or gcc)

**Comment by gbaraldi** [comment]
Right, but libtargetparser was part of libsupport, they just split it up for some reason (I guess it was kind of it's own thing,

**Comment by vtjnash** [comment]
Having libsupport needing to be linked in full is moderate large (1.5MB) relative to what is most useful from it for julia (about 8MB of this is julia debug info), so it also could end up being removed someday
```
10460   usr/lib/libjulia-internal.so.1.11.0 # with shared linking
12012   usr/lib/libjulia-internal.so.1.11.0 # with static linking
```

File: src/Makefile [hunk 1]
```diff
@@ -77,11 +77,8 @@ else
 # JULIACODEGEN != LLVM
 endif
 
-RT_LLVM_LIBS := support
-
-ifeq ($(LLVM_VER_MAJ),16)
-RT_LLVM_LIBS += targetparser
-endif
+RT_LLVM_LIBS := support # for APMath and some other useful ADT
+RT_LLVM_LIBS += targetparser # for getHostCPUName on LLVM 16+
```
Comment (giordano) [inline src/Makefile:81]: I think we still support llvm 15? CC @vchuravy.

Comment (vtjnash) [reply src/Makefile:81]: I guess I could invert the condition, to be ifneq 15?

Comment (giordano) [reply src/Makefile:81]:
```suggestion
ifneq ($(LLVM_VER_MAJ),15)
RT_LLVM_LIBS += targetparser # for getHostCPUName on LLVM 16+
endif
```

**Pull request merged**